### ISSUE TITLE
"Filename pattern" not clear

### ISSUE THREAD

**Issue by cool-RR** [thomasschafer/scooter#34]
I was using "Filename pattern" for the first time and it wasn't clear to me whether it captures the whole path or just the file name. If I understand correctly it's capturing only the file name, which was a pity for me because I wanted to use Scooter for two folders. Now I have to make two scooter runs.

**Comment by thomasschafer** [comment]
I should rename that field to something like "Path pattern" rather than "Filename pattern" as captures the whole path to the file, relative to the directory being searched in. If you run in to any issues with it let me know, and I'll update the field name and add to the documentation

**Comment by cool-RR** [comment]
A couple more question I have as a user when using this field:

1. Will the `./` at the beginning be included?
2. On Windows in zsh/bash, would it use forward slashes or backslashes?
3. Will it be doing a regex match, or a full match, or a search?

Maybe there should be some kind of help screen for regex that clarifies these points.

**Comment by thomasschafer** [comment]
> A couple more question I have as a user when using this field:
> 
> 1. Will the `./` at the beginning be included?
> 2. On Windows in zsh/bash, would it use forward slashes or backslashes?
> 3. Will it be doing a regex match, or a full match, or a search?
> 
> Maybe there should be some kind of help screen for regex that clarifies these points.

1. No, e.g. if you are searching in `/foo/` then a file `/foo/bar/qux.txt` will appear as `bar/qux.txt` when being matched against
2. I don't use Windows so I can't say for certain but this should be the platform default, so presumably backslashes (although note that these will need to be escaped because the field uses regex)
3. Regex (this is shown in the field name, I just didn't include it above for brevity)

**Comment by cool-RR** [comment]
I don't understand your answer to the third question.

**Comment by cool-RR** [comment]
To clarify, I'm asking whether it's a regex match, regex full match or regex search:

https://docs.python.org/3/library/re.html#re.match
https://docs.python.org/3/library/re.html#re.fullmatch
https://docs.python.org/3/library/re.html#re.search

**Comment by thomasschafer** [comment]
> To clarify, I'm asking whether it's a regex match, regex full match or regex search:
> 
> https://docs.python.org/3/library/re.html#re.match https://docs.python.org/3/library/re.html#re.fullmatch https://docs.python.org/3/library/re.html#re.search

I'm using [this function](https://docs.rs/regex/latest/regex/struct.Regex.html#method.is_match), which appears to correspond with `re.search` from your list, but there may be slight variations in behaviour so worth double checking the Rust docs
